business-registry-dashboard/app/pages/[encodedOrgId]/affiliationInvitation/acceptToken/[token].vue

```diff
@@ -0,0 +1,86 @@
+<script setup lang="ts">
+import { useRoute } from 'vue-router'
+
+const { t } = useI18n()
+const accountStore = useConnectAccountStore()
+const affStore = useAffiliationsStore()
+const brdModal = useBrdModals()
+const route = useRoute()
+const toast = useToast()
+
+// Token parsing
+const parseToken = (encodedToken: string): AffiliationToken => {
+  try {
+    const tokenObject = encodedToken.split('.')[0]
+    const decoded = atob(tokenObject as string)
+    return JSON.parse(decoded)
+  } catch (error) {
+    console.error('Failed to parse token:', error)
+    throw new Error('Invalid token format')
+  }
+}
+
+definePageMeta({
+  checkMagicLink: true,
+  order: 0
+})
+
+onMounted(async () => {
+  try {
+    const token = parseToken(route.params.token as string)
+    const orgId = token.fromOrgId
+    if (orgId === accountStore.currentAccount.id) { // Check if the account is the same as the one that initiated the request
```

Comment:
This is to handle situations where:

I request on account X to manage business Z. I log out and sign in, in account Y. Now if I press on the magic link, A toast will show `Please login with the account that initiated the request.`